inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -280,33 +280,44 @@ def fix_submission_number(obj, eng):
 
 
 @with_debug_logging
-def submission_fulltext_download(obj, eng):
+def populate_submission_document(obj, eng):
     submission_pdf = obj.extra_data.get('submission_pdf')
     if submission_pdf and is_pdf_link(submission_pdf):
         filename = secure_filename('fulltext.pdf')
-        pdf = download_file_to_workflow(
-            workflow=obj,
-            name=filename,
+        obj.data['documents'] = [
+            document for document in obj.data.get('documents', ())
+            if document.get('key') != filename
+        ]
+        lb = LiteratureBuilder(
+            source=obj.data['acquisition_source']['source'], record=obj.data)
+        lb.add_document(
+            filename,
+            fulltext=True,
             url=submission_pdf,
+            original_url=submission_pdf,
         )
+        obj.data = lb.record
 
-        if pdf:
-            obj.data['documents'] = [
-                document for document in obj.data.get('documents', ())
-                if document.get('key') != filename
-            ]
-            lb = LiteratureBuilder(source=obj.data['acquisition_source']['source'], record=obj.data)
-            lb.add_document(
-                filename,
-                fulltext=True,
-                original_url=submission_pdf,
-                url='/api/files/{bucket}/{key}'.format(bucket=obj.files[filename].bucket_id, key=filename)
+
+@with_debug_logging
+def download_documents(obj, eng):
+    documents = obj.data.get('documents')
+    if documents:
+        for document in documents:
```

Comment:
for future reference, a neater way to do this is to use `get('documents', [])`, which allows one not to have an `if`, but instead iterate directly on the (potentially empty) list.